### PR TITLE
FP-285 FallbackVastTags may not be working

### PR DESCRIPTION
**Pull request by Ankit-Exads**

### CODE REVIEW

**Review by ZentoDH — COMMENTED** [review]
Just tested this in your branch and the issue still persists. Could you make a dev summary with your test method?

**Review by fpfcarvalho — APPROVED** [review]

File: src/modules/adsupport.js [hunk 1]
```diff
@@ -1615,6 +1615,9 @@ export default function (playerInstance, options) {
     };
 
     playerInstance.decreaseSkipOffset = () => {
+        if (typeof playerInstance.vastOptions === 'undefined' || playerInstance.vastOptions === null) {
```
Comment (fpfcarvalho) [inline src/modules/adsupport.js:1618]: what happens if they are empty? it will break anyway

**Review by fpfcarvalho — CHANGES_REQUESTED** [review]

File: src/modules/adsupport.js (hunk 1 above)
Comment (ZentoDH) [reply src/modules/adsupport.js:1618]: This ticket can't be fixed by testing if something is undefined. You'll have to take a look at the try catch methods to see what the errors are, and based on that make a fix. The vastOptions could have data in it, but the data it gets from the server might be invalid xml, a 404/500/..., or maybe even not a xml at all.

Comment (Ankit-Exads) [reply src/modules/adsupport.js:1618]: Updated the code

File: src/modules/vast.js [hunk 2]
```diff
@@ -772,13 +772,69 @@ export default function (playerInstance, options) {
...
+                let mediaFileUrl = '';
+                if (Array.from(adElement.getElementsByTagName('AdParameters')).length) {
+                    const mediaFiles = Array.from(adElement.getElementsByTagName('AdParameters'));
+                    mediaFileIsValid = false;
+                    for (const mediaFile of mediaFiles) {
+                        mediaFileUrl = mediaFile.textContent.trim();
+                        try {
+                            const mediaFileObj = JSON.parse(mediaFileUrl);
+                            mediaFileUrl = mediaFileObj.videos[0].url;
+                            console.log("Extracted video URL:", mediaFileUrl);
```
Comment (ZentoDH) [inline src/modules/vast.js:785]: There's still a console log here

File: src/modules/vast.js [hunk 3]
```diff
@@ -772,13 +772,69 @@ export default function (playerInstance, options) {
...
+                            console.error("Error parsing media file URL:", error);
+                        }
+                    }
+                } else if (Array.from(adElement.getElementsByTagName('MediaFiles')).length) {
+                    const mediaFiles = Array.from(adElement.getElementsByTagName('MediaFiles'));   
+                    const mediaFile = mediaFiles[0].getElementsByTagName('MediaFile');
+                    mediaFileIsValid = false;
+                    for (const mediaFileTemp of mediaFile) {
+                        mediaFileUrl = mediaFileTemp.textContent.trim();
+                        console.log("Extracted video URL:", mediaFileUrl);
```
Comment (ZentoDH) [inline src/modules/vast.js:796]: Same here, console.log still here

**Review by ZentoDH — COMMENTED** [review]
Just tested the code and it seems like it's still just error'ing out and playing the video instead of going to the fallback. Maybe we should have a call to go over it together and look a bit deeper into it.

File: src/modules/vast.js (hunk 2 above)
Comment (Ankit-Exads) [reply src/modules/vast.js]: removed console.log

**Review by ZentoDH — APPROVED** [review]

**Review by fpfcarvalho — APPROVED** [review]

**Pull request merged**